proposals/csharp-7.0/tuples.md

```diff
@@ -0,0 +1,437 @@
+
+> note:
+>
+> For the purpose of overloading, overriding and hiding, tuples of the same types and lengths as well as their underlying ValueTuple types are considered equivalent. All other differences are immaterial. When overriding a member it is permitted to use tuple types with same or different field names than in the base member.
+>
+> A situation where same field names are used for non-matching fields between base and derived member signatures, a warning is reported by the compiler.  
+
+```csharp
+class Base
+{
+    virtual void M1(ValueTuple<int, int> arg){...}
+}
+class Derived : Base
+{
+    override void M1((int c, int d) arg){...} // valid override, signatures are equivalent
+}
+class Derived2 : Derived 
+{
+    override void M1((int c1, int c) arg){...} // also valid, warning on possible misuse of name 'c' 
+}
+
+class InvalidOverloading 
+{
+    virtual void M1((int c, int d) arg){...}
+    virtual void M1((int x, int y) arg){...}        // invalid overload, signatures are eqivalent
+    virtual void M1(ValueTuple<int, int> arg){...}  // also invalid
+}
+```
+
+> endnote.
+
+### Tuple field name erasure at runtime
+
+Tuple field names aren't part of the runtime representation of tuples, but are tracked only by the compiler. As a result, the field names will not be available to a 3rd party observer of a tuple instance - such as reflection or dynamic code.
+
+In alignment with the identity conversions, a boxed tuple does not retain the names of the fields and will unbox to any tuple type that has the same element types in the same order.
+
+```csharp
+object o = (a: 1, b: 2);           // boxing conversion
+var t = ((int moo, int boo))o;     // unboxing conversion
+```
+
+## Additions to [Variables](../../spec/variables.md)
+
+> The following text should be added at the end of the [Variables](../../spec/variables.md) section.
+
+### "Discards"
+
+The `_` can be used as a *discard* under these rules:
+
+- A standalone `_` is a discard when no `_` is defined in scope.
+- A "designator" `var _` or `T _` in deconstruction, pattern matching and out vars is a discard.
+- Discards are like unassigned variables, and do not have a value. They can only occur in contexts where they are assigned to.
+
+Examples:
+
+```csharp
+M(out _, out var _, out int _); // three out variable discards
+(_, var _, int _) = GetCoordinates(); // deconstruction into discards
+if (x is var _ && y is int _) { ... } // discards in patterns
+```
+
+## Changes to [Conversions](../spec/conversions.md)
+
+> The following text should be added to [Identity conversions](../../spec/conversions.md#identity-conversions), after the bullet point on `object` and `dynamic`:
+
+* Element names are immaterial to tuple conversions. Tuples with the same types in the same order are identity convertible to each other or to and from corresponding underlying `ValueTuple` types, regardless of the names.
```

Comment:
>to and from corresponding underlying `ValueTuple` types [](start = 140, length = 55)

I think it needs to be said somewhere what "underlying type" means.  For example, the type `(int, int)` and `ValueTuple<int, int>` are defined to designate the *same* type.